File: app/src/main/java/com/edison/librolink/retrofit/GoogleBooksAPI.kt
package com.edison.librolink.retrofit

import retrofit2.http.GET
import retrofit2.http.Query

data class BookResponse(
    val items: List<BookItem>?
)

data class BookItem(
    val volumeInfo: VolumeInfo
)

data class VolumeInfo(
    val title: String,
    val authors: List<String>?,
    val description: String?,
    val imageLinks: ImageLinks?
)

interface GoogleBooksApi {
    @GET("volumes")
    suspend fun searchBooks(
        @Query("q") query: String,
        @Query("startIndex") startIndex: Int = 0, // Default to 0
        @Query("maxResults") maxResults: Int = 10 // Default to 10
    ): BookResponse
}
data class ImageLinks(
    val thumbnail: String?
)

File: lib/data/retrofit/google_books_api.dart
import 'package:dio/dio.dart';
import 'package:libro_link/data/retrofit/book_response.dart';
class GoogleBooksApi {
  final Dio _dio;

  GoogleBooksApi([Dio? dio])
      : _dio = dio ?? Dio(BaseOptions(baseUrl: 'https://www.googleapis.com/books/v1/'));

  Future<BookResponse> searchBooks({
    required String query,
    int startIndex = 0,
    int maxResults = 10,
  }) async {
    final response = await _dio.get('volumes', queryParameters: {
      'q': query,
      'startIndex': startIndex,
      'maxResults': maxResults,
    });

    return BookResponse.fromJson(response.data);
  }

  Future<BookItem> getBookById(String id) async {
    final response = await _dio.get('volumes/$id');
    return BookItem.fromJson(response.data);
  }
}

